Zend/tests/attributes/gh8421.phpt

```diff
@@ -0,0 +1,48 @@
+--TEST--
+Bug #8421: Attributes that target functions are not valid for anonymous functions defined within a method
+--FILE--
+<?php
+#[Attribute(Attribute::TARGET_FUNCTION)]
+class FunctionAttribute
+{
+    public int $number = 1;
+}
+
+$globalClosure       = #[FunctionAttribute] fn() => true;
```

Comment:
Not sure about writing the short function on the same line as the attribute; at least the syntax highlighting is confusing.

Reply:
Am happy to rectify this @cmb69, I went with that layout because adding the newline makes it just look a bit weird, but I get that's a personal preference. Will update.

```php
$globalClosure = #[FunctionAttribute]
fn() => true;
```

Reply:
I'm super unsure about this. Could also look like:
````php
$globalClosure = 
    #[FunctionalAttribute]
    fn () => true;
````